src/Uno.UI.RuntimeTests/Tests/Microsoft_UI_Xaml_Controls/Given_TreeView.cs

```diff
@@ -94,6 +95,104 @@ public async Task When_TreeViewItem_Dragged_Near_the_Edge()
 		}
 #endif
 
+		[TestMethod]
+#if __ANDROID__ || __IOS__
+		[Ignore("Fails")]
+#endif
+		public async Task When_TreeViewItem_Collapsed_Children_Removed_From_Tree()
+		{
+			var treeView = new TreeView
+			{
+				RootNodes =
+				{
+					new TreeViewNode
+					{
+						Content = "Parent",
+						IsExpanded = true,
+						Children =
+						{
+							new TreeViewNode
+							{
+								Content = "Child",
+								IsExpanded = true
+							}
+						}
+					}
+				}
+			};
+
+			await UITestHelper.Load(treeView);
+
+			Assert.AreEqual(2, treeView.FindVisualChildByType<ItemsStackPanel>().Children.Count);
+
+			treeView.RootNodes[0].IsExpanded = false;
+			await WindowHelper.WaitForIdle();
+
+			Assert.AreEqual(1, treeView.FindVisualChildByType<ItemsStackPanel>().Children.Count);
+
+			treeView.RootNodes[0].IsExpanded = true;
+			await WindowHelper.WaitForIdle();
+
+			Assert.AreEqual(2, treeView.FindVisualChildByType<ItemsStackPanel>().Children.Count);
+		}
+
+#if HAS_UNO
+		// https://github.com/unoplatform/uno/issues/16041
+		// This test actually always passes due to catching the NRE in SafeRaiseEvent in RaiseDragEnterOrOver
+		// so you need to check manually if there are any logged exceptions
```

Comment:
i think you can actually catch the exception to make the test pass/fail accordingly:
```cs
partial class TestableTreeView : TreeView
	public Action<DragEventArgs, Exception>? OnDragEnterExceptionHandler { get; set; }
	
	protected override void OnDragEnter(Microsoft.UI.Xaml.DragEventArgs args)
		try { base.OnDragEnter(args); }
		catch (Exception e) { OnDragEnterExceptionHandler?.Invoke(args, e); }
```